core/services/ocr2/plugins/ocr2keeper/evm21/registry.go

```diff
@@ -122,14 +127,26 @@ var upkeepStateEvents = []common.Hash{
 }
 
 type MercuryConfig struct {
-	cred *models.MercuryCredentials
-	abi  abi.ABI
+	Cred *models.MercuryCredentials
+	Abi  abi.ABI
 	// allowListCache stores the upkeeps privileges. In 2.1, this only includes a JSON bytes for allowed to use mercury
-	allowListCache *cache.Cache
+	AllowListCache *cache.Cache
+}
+
+func (c *MercuryConfig) Credentials() mercury.MercuryCredentials {
```

Comment:
If you can, try not to return interfaces. It makes the code more brittle. The ideal Go pattern is to consume interfaces and return structs.

Reply:
That is a trick used to avoid circular dependency. We have this `models.MercuryCredentials` struct defined in models package. 

```
type MercuryCredentials struct {
	LegacyURL string
	URL       string
	Username  string
	Password  string
}
```

We then defined this `mercury.MercuryCredentials` interface under mercury package. `models.MercuryCredentials` implements `mercury.MercuryCredentials` so we can use mercury in all mercury related references, while keep the other non-mercury references unchanged.

I inherited this from Fergal. Happy to learn more on how we can improve.